### PR TITLE
Handle an expired admin user link

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Adds an error page to the admin login flow. I added a layer above our `<App/>` that provides a way to add routes for things like addressable error pages. I've added `/error/expired-ots` for this scenario, but anything under `/error/*` will resolve to this new routing layer, and render a default error page if we don't have an explicit route for it yet. This will let us add more addressable error pages in the future if we need. The primary intent of these is for scenarios where the api layer needs to redirect somewhere in the case of an error, and let us render clientside to accomplish that. No external data is loaded from the client for this routing layer, such as a user/orgs.

image

## Related Issue(s)

Fixes WEB-286

## How to test

* You can manually view the error page by visiting `/error/expired-ots`
* Visiting `/error/foo` will render a default error page, mostly to just have a catch-all for any undefined routes underneath the `/error/*` path.
* Simulate a bad ots by visiting `/api/login/ots/admin/badtoken` - it should redirect to the `/error/expired-ots` error page

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-web-21aa1d24e1f</li>
	<li><b>🔗 URL</b> - <a href="https://brad-web-21aa1d24e1f.preview.gitpod-dev.com/workspaces" target="_blank">brad-web-21aa1d24e1f.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [x] with-dedicated-emulation
- [x] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/dashboard/src/App.tsx [hunk 1]
```diff
@@ -4,21 +4,16 @@
...
 import { useCurrentOrg } from "./data/organizations/orgs-query";
 import { useAnalyticsTracking } from "./hooks/use-analytics-tracking";
 import { useUserLoader } from "./hooks/use-user-loader";
 import { Login } from "./Login";
 import { AppBlockingFlows } from "./app/AppBlockingFlows";
-import { useHistory } from "react-router";
-
-// Wrap the App in an ErrorBoundary to catch User/Org loading errors
-// This will also catch any errors that happen to bubble all the way up to the top
-const AppWithErrorBoundary: FC = () => {
```
Comment (selfcontained) [inline components/dashboard/src/App.tsx:19]: This was leftover and not needed anymore.

File: components/dashboard/src/App.tsx [hunk 2]
```diff
@@ -72,4 +67,13 @@ const App: FC = () => {
     );
 };
 
-export default AppWithErrorBoundary;
+// Routing level above main App component for any routes that don't need user/orgs loaded, such as addressable error pages
```
Comment (selfcontained) [inline components/dashboard/src/App.tsx:70]: This adds a layer above our App to let us render things like addressable error pages w/ their own routes. They won't load any external data by default.

File: components/dashboard/src/login/SetupPending.tsx [hunk 3]
```diff
@@ -6,10 +6,10 @@
 
 import { FC } from "react";
 import { Heading2, Subheading } from "../components/typography/headings";
-import cubicleImg from "../images/cubicle.png";
-import cubicleDarkImg from "../images/cubicle-dark.png";
-import cubicleImg2x from "../images/cubicle@2x.png";
-import cubicleDarkImg2x from "../images/cubicle-dark@2x.png";
+import cubicleImg from "../images/cubicle.webp";
```
Comment (selfcontained) [inline components/dashboard/src/login/SetupPending.tsx:9]: Switched all these png to webp cause they're waaaaay smaller.

File: components/server/src/user/user-controller.ts [hunk 4]
```diff
@@ -184,10 +184,8 @@ export class UserController {
             } catch (e) {
                 log.error("Failed to sign-in as admin with OTS Token", e);
 
-                // Default to unauthenticated, to not leak information.
-                // We do not send the error response to ensure we do not disclose information.
-                const code = e.code || 401;
-                res.sendStatus(code);
+                // Always redirect to an expired token page if there's an error
+                res.redirect("/error/expired-ots", 307);
```
Comment (selfcontained) [inline components/server/src/user/user-controller.ts:188]: unsure if we want to try and classify different errors, i.e. expired vs. invalid. If so, we can have an error route for each one.

**Review by svenefftinge — APPROVED** [review]

**Pull request merged**